pintracker/stateless/stateless.go

```diff
@@ -224,11 +244,8 @@ func (spt *Tracker) Track(ctx context.Context, c *api.Pin) error {
 
 	logger.Debugf("tracking %s", c.Cid)
 
-	// Sharded pins are never pinned. A sharded pin cannot turn into
-	// something else or viceversa like it happens with Remote pins so
-	// we just track them.
+	// Sharded pins are never pinned, so do nothing.
 	if c.Type == api.MetaType {
-		spt.optracker.TrackNewOperation(ctx, c, optracker.OperationShard, optracker.PhaseDone)
```

Comment:
ok, be very careful. You are changing the functionality of the pin tracker. Do not do that and do not break it. This code is not here for fun.

Sharded pins need to be potentially reported in the Status calls and need to be tracked. And even if this was not the case, you are not supposed to change functionality on this PR.

Reply:
This operation is already in PhaseDone. What TrackNewOperation does in this case
- Create the operation with phase done and at it to operations map
- it is not added to any queue (pinCh or unpinCh), so there is no task to be performed.
-  few seconds later (ticker on every 10 sec), it gets cleaned because it is already in phase done.
Why add it, if it is going to get removed without doing anything?

Metatype pins are only there for the reference to cluster dag (right?), for which it doesn't need to get pinned. It only has to be present in the shared state.

Status calls only checks the state and returns status sharded.